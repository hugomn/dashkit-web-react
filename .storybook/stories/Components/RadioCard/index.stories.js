import React from "react";
import { storiesOf } from "@storybook/react";
import { Box, Heading, Text } from "grommet";
import RadioCardGroup from "../../../../src/components/RadioCardGroup";
import { action } from "@storybook/addon-actions";

storiesOf("Components/Radio Card Group", module).add("RadioCardGroup", () => {
  return (
    <Box>
      <Box>
        <Heading level="3">Radio Card Group</Heading>
        <RadioCardGroup
          options={[
            {
              value: "c1",
              label: "Choice 1",
              description:
                "Lorem ipsum dolor sit amet, ei usu recteque constituto, in vix iuvaret dolorum consequat. Id quis dicat civibus sit, ut vis meis dissentiet, ad sale dignissim per. Ea iusto errem omittam mei, viderer habemus."
            },
            {
              value: "c2",
              label: "Choice 2",
              description:
                "Lorem ipsum dolor sit amet, ei usu recteque constituto, in vix iuvaret dolorum consequat. Id quis dicat civibus sit, ut vis meis dissentiet, ad sale dignissim per. Ea iusto errem omittam mei, viderer habemus."
            },
            {
              value: "c3",
              label: "Choice 3",
              description:
                "Lorem ipsum dolor sit amet, ei usu recteque constituto, in vix iuvaret dolorum consequat. Id quis dicat civibus sit, ut vis meis dissentiet, ad sale dignissim per. Ea iusto errem omittam mei, viderer habemus."
            }
          ]}
          onSelect={value => alert(value)}
          selectedValue="c2"
        />
      </Box>

      <Box>
        <Heading level="3">Disabled Radio Cards</Heading>
        <RadioCardGroup
          disabled
          options={[
            {
              value: "c1",
              label: "Choice 1",
              description:
                "Lorem ipsum dolor sit amet, ei usu recteque constituto, in vix iuvaret dolorum consequat. Id quis dicat civibus sit, ut vis meis dissentiet, ad sale dignissim per. Ea iusto errem omittam mei, viderer habemus."
            },
            {
              value: "c2",
              label: "Choice 2",
              description:
                "Lorem ipsum dolor sit amet, ei usu recteque constituto, in vix iuvaret dolorum consequat. Id quis dicat civibus sit, ut vis meis dissentiet, ad sale dignissim per. Ea iusto errem omittam mei, viderer habemus."
            },
            {
              value: "c3",
              label: "Choice 3",
              description:
                "Lorem ipsum dolor sit amet, ei usu recteque constituto, in vix iuvaret dolorum consequat. Id quis dicat civibus sit, ut vis meis dissentiet, ad sale dignissim per. Ea iusto errem omittam mei, viderer habemus."
            }
          ]}
        />
      </Box>
    </Box>
  );
});
