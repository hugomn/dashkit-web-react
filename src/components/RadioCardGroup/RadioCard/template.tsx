import { Box, RadioButton, Text } from "grommet";
import * as React from "react";
import styled from "styled-components";
import theme from "../../../constants/theme";

interface IProps {
  value: string;
  label: string;
  description: string;
  disabled: boolean;
  selected: boolean;
  onSelect: any;
}

const RadioCard: React.FC<IProps> = ({ selected, onSelect, disabled, ...props }) => {
  const { value, label, description } = props;
  return (
    <StyledBox
      data-testid="choice"
      disabled={disabled}
      pad="medium"
      selected={selected}
      onClick={() => !disabled && onSelect(value)}
      background="white"
      elevation="xsmall">
      <RadioButton
        data-testid={value}
        disabled={disabled}
        label={label}
        name="radio"
        value={value}
        checked={selected}
        onChange={(e: any) => onSelect(e.target.value)}
      />
      <Box pad={{ top: "small" }}>
        <Text>{description}</Text>
      </Box>
    </StyledBox>
  );
};

interface IStyledBoxProps {
  selected: boolean;
  disabled: boolean;
}

const StyledBox = styled(Box)`
  ${({ selected, disabled }: IStyledBoxProps) => `
  border-radius: 2px;
  height: 100%;
  label {
    font-weight: bold;
    opacity: 1;
  }
  transition: all 0.1s ease;

  ${
    selected
      ? `
    border: 1px solid ${theme.global.colors.brand};
    box-shadow: 0 0 2px ${theme.global.colors.brand};
    label {
      color: ${theme.global.colors.brand};
    }
  `
      : `
    border: 1px solid white;    
  `
  }

  ${disabled &&
    !selected &&
    `
    background-color: ${theme.global.colors["light-3"]};
    border: 1px solid ${theme.global.colors["light-3"]};
    color: ${theme.global.colors["dark-3"]};    
  `}

  ${!disabled &&
    `
    label {
      color: ${theme.global.colors.brand};
    }
    &:hover {
      cursor: pointer;
      label {
        color: ${theme.global.colors.brand};
      }
    }
    `}
  `}
`;

export default RadioCard;
