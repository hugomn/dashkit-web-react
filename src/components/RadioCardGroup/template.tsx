import { Box } from "grommet";
import * as React from "react";
import RadioCard from "./RadioCard/template";

interface IOptionProps {
  value: string;
  label: string;
  description: string;
}

interface IProps {
  disabled?: boolean;
  options: IOptionProps[];
  onSelect?: (value: string) => void;
  value?: string;
}

export const handleSelect = (onSelect: any) => (value: string) => {
  onSelect(value);
};

const RadioCardGroup: React.FC<IProps> = ({ disabled = false, options, onSelect, value }) => {
  return (
    <Box
      pad={{ horizontal: "medium" }}
      direction="row-responsive"
      align="stretch"
      data-testid="radioCardGroup">
      {options.map((option, i, { length }) => {
        return (
          <Box
            basis={`${100 / length}%`}
            key={option.value}
            margin={{ right: length - 1 !== i ? "medium" : "", top: "medium" }}>
            <RadioCard
              disabled={disabled}
              value={option.value}
              selected={value === option.value}
              onSelect={handleSelect(onSelect)}
              label={option.label}
              description={option.description}
            />
          </Box>
        );
      })}
    </Box>
  );
};

export default RadioCardGroup;
