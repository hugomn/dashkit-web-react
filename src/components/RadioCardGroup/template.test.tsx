import { cleanup, fireEvent, render } from "@testing-library/react";
import { shallow } from "enzyme";
import * as React from "react";
import RadioCard from "./RadioCard";
import RadioCardGroup from ".";

const options = [
  {
    value: "c1",
    label: "Choice 1",
    description: "Lorem ipsum dolor sit amet, ei usu recteque constituto."
  },
  {
    value: "c2",
    label: "Choice 2",
    description: "Lorem ipsum dolor sit amet, ei usu recteque constituto."
  },
  {
    value: "c3",
    label: "Choice 3",
    description: "Lorem ipsum dolor sit amet, ei usu recteque constituto."
  }
];

describe("Radio Card Group", () => {
  afterEach(() => cleanup());

  it("Render a radio card group", () => {
    const wrapper = shallow(<RadioCardGroup options={options} />);
    expect(wrapper.find(RadioCard)).toHaveLength(options.length);
  });

  it("Displays a disabled card group", () => {
    const { container } = render(<RadioCardGroup disabled options={options} />);
    expect(container).toMatchSnapshot();
  });

  it("Calls onSelect", () => {
    const mockOnSelect = jest.fn();
    const { getByDisplayValue } = render(
      <RadioCardGroup disabled options={options} onSelect={mockOnSelect} />
    );
    fireEvent.click(getByDisplayValue("c1"));
    expect(mockOnSelect).toHaveBeenCalled();
  });
});
