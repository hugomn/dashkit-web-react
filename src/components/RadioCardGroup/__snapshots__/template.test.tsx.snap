// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Card Group Displays a disabled card group 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 gmRjHO"
    data-testid="radioCardGroup"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 bZTTyO"
    >
      <div
        class="sc-bdVaJa bUODpI StyledBox-sc-13pk1d4-0 erxBvF"
        data-testid="choice"
        disabled=""
      >
        <label
          class="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hgwIsh StyledBox-sc-13pk1d4-0 gQLnBt"
          disabled=""
        >
          <div
            class="StyledRadioButton-g1f6ld-4 gVogic StyledBox-sc-13pk1d4-0 kvuUVC"
          >
            <input
              class="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 bldEoB"
              data-testid="c1"
              disabled=""
              name="radio"
              type="radio"
              value="c1"
            />
            <div
              class="StyledRadioButton__StyledRadioButtonBox-g1f6ld-3 lnEyXE StyledBox-sc-13pk1d4-0 ghGSYb"
            />
          </div>
          <span>
            Choice 1
          </span>
        </label>
        <div
          class="StyledBox-sc-13pk1d4-0 kFozry"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Lorem ipsum dolor sit amet, ei usu recteque constituto.
          </span>
        </div>
      </div>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 bZTTyO"
    >
      <div
        class="sc-bdVaJa bUODpI StyledBox-sc-13pk1d4-0 erxBvF"
        data-testid="choice"
        disabled=""
      >
        <label
          class="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hgwIsh StyledBox-sc-13pk1d4-0 gQLnBt"
          disabled=""
        >
          <div
            class="StyledRadioButton-g1f6ld-4 gVogic StyledBox-sc-13pk1d4-0 kvuUVC"
          >
            <input
              class="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 bldEoB"
              data-testid="c2"
              disabled=""
              name="radio"
              type="radio"
              value="c2"
            />
            <div
              class="StyledRadioButton__StyledRadioButtonBox-g1f6ld-3 lnEyXE StyledBox-sc-13pk1d4-0 ghGSYb"
            />
          </div>
          <span>
            Choice 2
          </span>
        </label>
        <div
          class="StyledBox-sc-13pk1d4-0 kFozry"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Lorem ipsum dolor sit amet, ei usu recteque constituto.
          </span>
        </div>
      </div>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 iEPtuR"
    >
      <div
        class="sc-bdVaJa bUODpI StyledBox-sc-13pk1d4-0 erxBvF"
        data-testid="choice"
        disabled=""
      >
        <label
          class="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 hgwIsh StyledBox-sc-13pk1d4-0 gQLnBt"
          disabled=""
        >
          <div
            class="StyledRadioButton-g1f6ld-4 gVogic StyledBox-sc-13pk1d4-0 kvuUVC"
          >
            <input
              class="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 bldEoB"
              data-testid="c3"
              disabled=""
              name="radio"
              type="radio"
              value="c3"
            />
            <div
              class="StyledRadioButton__StyledRadioButtonBox-g1f6ld-3 lnEyXE StyledBox-sc-13pk1d4-0 ghGSYb"
            />
          </div>
          <span>
            Choice 3
          </span>
        </label>
        <div
          class="StyledBox-sc-13pk1d4-0 kFozry"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Lorem ipsum dolor sit amet, ei usu recteque constituto.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
